pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -619,6 +726,25 @@ def _get_all_annotations(node: nodes.FunctionDef) -> list[nodes.NodeNG | None]:
     return annotations
 
 
+def _get_kw_annotation(
+    node: nodes.FunctionDef, key: str
+) -> tuple[nodes.NodeNG | None, nodes.NodeNG | None]:
+    args = node.args
+    for index, arg_name in enumerate(args.posonlyargs):
+        if key == arg_name.name:
+            return arg_name, args.posonlyargs_annotations[index]
```

Comment:
The function name is a bit misleading if you also check for `posonlyargs_annotations`.
In practice, checking just `annotations` and `kwonlyargs_annotations` should be enough.

Reply:
Yep, that works. I've adjusted accordingly.